nym-vpn-core/nym-vpn-lib/src/lib.rs

```diff
@@ -16,832 +16,35 @@ mod routing;
 mod tunnel;
 mod tunnel_setup;
 mod uniffi_custom_impls;
+mod vpn;
 mod wg_gateway_client;
 mod wireguard_setup;
 
-use std::{
-    net::{IpAddr, Ipv4Addr},
-    path::PathBuf,
-    sync::{Arc, Mutex},
-};
-
-use futures::{
-    channel::{mpsc, oneshot},
-    SinkExt,
-};
-use log::{debug, error, info};
-use nym_connection_monitor::ConnectionMonitorTask;
-use nym_gateway_directory::{
-    Config as GatewayDirectoryConfig, EntryPoint, ExitPoint, GatewayClient, IpPacketRouterAddress,
-};
-use nym_ip_packet_client::IprClientConnect;
-use nym_task::TaskManager;
-use talpid_core::dns::DnsMonitor;
-use talpid_routing::RouteManager;
-use talpid_tunnel::tun_provider::TunProvider;
-use tokio::task::JoinHandle;
-use tun2::AsyncDevice;
-use tunnel_setup::{init_firewall_dns, setup_tunnel, AllTunnelsSetup, TunnelSetup};
-use util::{wait_and_handle_interrupt, wait_for_interrupt_and_signal};
-
-#[cfg(target_os = "ios")]
-use crate::platform::swift::OSTunProvider;
-use crate::{
-    error::Result,
-    mixnet::SharedMixnetClient,
-    uniffi_custom_impls::{ExitStatus, StatusEvent},
-};
-
-// Public re-export some dependencies
-pub use nym_bin_common;
+// Re-export some our nym dependencies
+pub use nym_bin_common as bin_common;
```

Comment:
Do we use `nym_bin_common` in vpnlib?

Reply:
yep in a few places. In platform and mixnet connect